From e04ba60f7d8c67c8f05aca38744a9ef4b2b7ff0c Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 29 Aug 2025 11:15:33 +0300
Subject: [PATCH 100/N] build: Add ABIFLAGS to targets when needed

So we get different names for debug and free threaded builds
---
 Makefile.pre.in | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index f3236eb..d35bb1a 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -315,11 +315,11 @@ MACHDEP_OBJS=	@MACHDEP_OBJS@
 LIBOBJDIR=	Python/
 LIBOBJS=	@LIBOBJS@
 
-PYTHON=		python$(EXE)
-BUILDPYTHON=	python$(BUILDEXE)
-BUILDPYTHONW=	pythonw$(BUILDEXE)
-BUILDVENVLAUNCHER=	venvlauncher$(BUILDEXE)
-BUILDVENVWLAUNCHER=	venvwlauncher$(BUILDEXE)
+PYTHON=		python$(ABIFLAGS)$(EXE)
+BUILDPYTHON=	python$(ABIFLAGS)$(BUILDEXE)
+BUILDPYTHONW=	pythonw$(ABIFLAGS)$(BUILDEXE)
+BUILDVENVLAUNCHER=	venvlauncher$(ABIFLAGS)$(BUILDEXE)
+BUILDVENVWLAUNCHER=	venvwlauncher$(ABIFLAGS)$(BUILDEXE)
 
 HOSTRUNNER= @HOSTRUNNER@
 
@@ -2484,7 +2484,7 @@ altbininstall: $(BUILDPYTHON) @FRAMEWORKPYTHONW@
 	done
 	if test "$(PYTHONFRAMEWORKDIR)" = "no-framework" ; then \
 		$(INSTALL_PROGRAM) $(BUILDPYTHON) $(DESTDIR)$(BINDIR)/python$(LDVERSION)$(EXE); \
-		$(INSTALL_PROGRAM) $(BUILDPYTHONW) $(DESTDIR)$(BINDIR)/python3w$(EXE); \
+		$(INSTALL_PROGRAM) $(BUILDPYTHONW) $(DESTDIR)$(BINDIR)/python3w$(ABIFLAGS)$(EXE); \
 		$(INSTALL_PROGRAM) $(BUILDPYTHONW) $(DESTDIR)$(BINDIR)/python$(LDVERSION)w$(EXE); \
 		$(INSTALL_PROGRAM) $(BUILDVENVLAUNCHER) $(DESTDIR)$(VENVLAUNCHERDIR)/$(BUILDVENVLAUNCHER); \
 		$(INSTALL_PROGRAM) $(BUILDVENVWLAUNCHER) $(DESTDIR)$(VENVLAUNCHERDIR)/$(BUILDVENVWLAUNCHER); \
